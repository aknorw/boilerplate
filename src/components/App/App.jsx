import React from 'react'

import Button from 'components/Button'

const App = () => (
  <div>
    <h1>
      Hello React
    </h1>
    <Button onClick={() => alert('Clicked')}>
      Click me
    </Button>
  </div>
)

export default App
