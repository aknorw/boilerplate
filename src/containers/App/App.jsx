import React from 'react'
import PropTypes from 'prop-types'
import { Helmet } from 'react-helmet'
import { connect } from 'react-redux'
import { createStructuredSelector } from 'reselect'

import { setFoo } from 'services/basic/actions'
import { selectFoo } from 'services/basic/selectors'

import Main from './components/Main'

export const App = ({ foo, handleButtonClick }) => (
  <div>
    <Helmet
      titleTemplate={`%s - ${APP_NAME}`}
      defaultTitle={APP_NAME}
    >
      {/* The following line will be used when implementing i18n */}
      <html lang="en" />
    </Helmet>
    <Main
      currentValue={foo}
      handleButtonClick={handleButtonClick}
    />
  </div>
)

App.propTypes = {
  foo: PropTypes.string.isRequired,
  handleButtonClick: PropTypes.func.isRequired,
}

const mapStateToProps = createStructuredSelector({
  foo: selectFoo(),
})

const mapDispatchToProps = dispatch => ({
  handleButtonClick: () => dispatch(setFoo()),
})

const withConnect = connect(mapStateToProps, mapDispatchToProps)

export default withConnect(App)
