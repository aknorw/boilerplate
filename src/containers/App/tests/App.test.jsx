import React from 'react'
import { shallow } from 'enzyme'

import { App } from '../App'

const mockFn = jest.fn()
console.error = jest.fn()

describe('<App />', () => {
  it('should be defined', () => {
    expect(App).toBeDefined()
  })
  it('should log 4 errors when no props are passed', () => { // 2 for App, 2 for Main
    shallow(<App />)
    expect(console.error).toHaveBeenCalledTimes(4)
  })
  it('should render correctly', () => {
    const foo = 'bar'
    const tree = shallow(
      <App
        foo={foo}
        handleButtonClick={mockFn}
      />,
    )
    expect(tree).toMatchSnapshot()
  })
})
