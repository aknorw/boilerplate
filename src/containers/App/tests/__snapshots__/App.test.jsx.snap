// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render correctly 1`] = `
<div>
  <HelmetWrapper
    defaultTitle="JEST"
    defer={true}
    encodeSpecialCharacters={true}
    titleTemplate="%s - JEST"
  >
    <html
      lang="en"
    />
  </HelmetWrapper>
  <Main
    currentValue="bar"
    handleButtonClick={[MockFunction]}
  />
</div>
`;
